sw/device/silicon_creator/rom/e2e/rom_ext_upgrade_test.c

```diff
@@ -0,0 +1,30 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/boot_data.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+#include "sw/device/silicon_creator/lib/manifest_def.h"
+
+#include "otp_ctrl_regs.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// What's using this return value?
+bool test_main(void) {
+  boot_data_t boot_data;
+  RETURN_IF_ERROR(boot_data_read(kLcStateProd, &boot_data));
+  LOG_INFO("Ver:%x", boot_data.min_security_version_rom_ext);
+  boot_data.min_security_version_rom_ext++;
+  RETURN_IF_ERROR(boot_data_write(&boot_data));
+  RETURN_IF_ERROR(boot_data_read(kLcStateProd, &boot_data));
+  LOG_INFO("Ver:%x", boot_data.min_security_version_rom_ext);
```

Comment:
We can remove this since we will print the version after reset anyway. (write and version check above should depend on the reset reason)

Reply:
I haven't written any flash code to run after reset. I just see a BFV ROM and LCV message after reset

Reply:
Do we need it to check reset_reason? I've set it to fail if it logs to UART either a version of 3 because it can update and persist but isn't stopping the rom_ext from executing or if it sees version 0 more than once because it can't update or can't persist.

Reply:
There's also a timeout to enforce that either of these would count as a failure.

Reply:
I think what you have right now is fine. Thanks @drewmacrae !